pkg/platform/kube/platform.go

```diff
@@ -1792,7 +1792,9 @@ func (p *Platform) validateAPIGatewayMeta(platformAPIGatewayMeta *platform.APIGa
 	errorMessages := validation.IsQualifiedName(platformAPIGatewayMeta.Name)
 	if len(errorMessages) != 0 {
 		joinedErrorMessage := strings.Join(errorMessages, ", ")
-		return errors.Errorf("Api gateway name doesn't conform to k8s naming convention. Errors: %s", joinedErrorMessage)
+		return nuclio.NewErrBadRequest(fmt.Sprintf(
+			"Api gateway name doesn't conform to k8s naming convention. Errors: %s",
+			joinedErrorMessage))
```

Comment:
could you please validate how the `joinedErrorMessage` looks?
In the current implementation (without this PR) there are many duplicate messages:
```
{
    "error": "Api gateway name doesn't conform to k8s naming convention. Errors: name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')",
    "errorStackTrace": "\nError - Api gateway name doesn't conform to k8s naming convention. Errors: name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')\n    /nuclio/pkg/platform/kube/platform.go:1795\n\nCall stack:\nApi gateway name doesn't conform to k8s naming convention. Errors: name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')\n    /nuclio/pkg/platform/kube/platform.go:1795\nFailed to validate API-gateway metadata\n    /nuclio/pkg/platform/kube/platform.go:1823\nFailed to validate and enrich an API-gateway name\n    /nuclio/pkg/platform/kube/platform.go:842\n"
}
```

Reply:
@TomerShor here it is:

```
name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')
```